frontend/src/app/application-forms/tree-application-form/tree-application-form.component.ts

```diff
@@ -0,0 +1,100 @@
+import { Title } from '@angular/platform-browser';
+import { Component, OnInit, ViewEncapsulation } from '@angular/core';
+import { alphanumericValidator } from '../validators/alphanumeric-validation';
+import { FormBuilder, Validators, FormGroup } from '@angular/forms';
+import { ActivatedRoute } from '@angular/router';
+import { currencyValidator } from '../validators/currency-validation';
+import { lessThanOrEqualValidator } from '../validators/less-than-or-equal-validation';
+import { TreesService } from '../../trees/_services/trees.service';
+import { ApplicationFieldsService } from '../_services/application-fields.service';
+import { ChristmasTreesApplicationService } from '../../trees/_services/christmasTreesApplication.service';
+
+@Component({
+  selector: 'app-tree-application-form',
+  templateUrl: './tree-application-form.component.html'
+})
+export class TreeApplicationFormComponent implements OnInit {
+  forest: any;
+  submitted = false;
+  application: any;
+  applicationForm: FormGroup;
+  maxNumberOfTrees: number;
+  quantityLength: number;
+  costPerTree: number;
+  apiErrors: any;
+
+  constructor(
+    private route: ActivatedRoute,
+    private titleService: Title,
+    public formBuilder: FormBuilder,
+    public applicationService: ChristmasTreesApplicationService,
+    public applicationFieldsService: ApplicationFieldsService,
+    private treesService: TreesService
+  ) {
+    this.applicationForm = this.formBuilder.group({
+      forestId: ['', [Validators.required]],
+      orgStructureCode: ['', [Validators.required]],
+      treeCost: [''],
```

Comment:
is there no validation for this because its supplied by our data? is there an injection concern with this?

Reply:
Yes, no validation because we supply the data, and the fields are not available in the form. It is open to injection, but it's probably not something that we can prevent. The forestId is a foreign key in the table, so if the id doesn't match one of our forests, the request will error. @lauraGgit